src/studio/src/designer/backend/Infrastructure/GitRepository/AltinnAppGitRepository.cs

```diff
@@ -185,6 +187,38 @@ public async Task<Designer.Models.TextResource> GetTextV1(string language)
             return textResource;
         }
 
+        /// <summary>
+        /// Returns content in text file written in the new text format
+        /// identified by languageCode in filename as a string.
+        /// </summary>
+        /// <param name="languageCode">Language identifier</param>
+        public async Task<string> GetTextContentV2(string languageCode)
+        {
+            string fileName = "text." + languageCode + ".json";
+
+            var textFileRelativeFilePath = Path.Combine(CONFIG_FOLDER_PATH, LANGUAGE_RESOURCE_FOLDER_NAME, fileName);
+
+            string fileContent = await ReadTextByRelativePathAsync(textFileRelativeFilePath);
```

Comment:
`ReadTextByRelativePathAsync` could possibly throw an IOException if the file does not exists. 
We should probably handle the IOException and return an empty dictionary or null.
If the controller should return 404 if the language does not exists we should return null, otherwise we should return an empty Dictionary

Reply:
This is fixed in a separate issue #8921 

@tjololo Great if you could take a look 😇

Reply:
great, closing this as it will be resolved in another issue :)